Marlin/src/gcode/gcode.cpp

```diff
@@ -823,6 +823,10 @@ void GcodeSuite::process_next_command() {
    * G-code "macros" to be called from within other G-code handlers.
    */
 
+  #if !defined(__AVR__) && !defined(strchr_P) // seems required on STM32 platform
```

Comment:
STM MCUs don't have a concept of PGM space as something you need to reference differently, so it's not surprising the macros are missing. Is there a reason you need the __AVR__ clause? It seems like on AVR strchr_p would be defined.

Reply:
git grep -n strchr_p

Reply:
The AVR PGM version of functions should be aliased to the standard ones within the platform (`HAL.h` more than likely) if not done directly within the architectures Arduino framework as it should be.

Reply:
yes, 
#ifdef __AVR__
at least.. seems the same function with some memory attribute (for this case)